com.unity.shadergraph/Editor/Data/Nodes/AbstractMaterialNode.cs

```diff
@@ -33,6 +33,9 @@ abstract class AbstractMaterialNode : JsonObject, IGroupItem, IRectInterface
         [NonSerialized]
         bool m_IsActive = true;
 
+        [NonSerialized]
+        bool m_WasUsedByGenerator = false;
```

Comment:
The easiest way to figure out whether errors being generated were on "used" nodes or not was to flag the nodes used during import, then use the flags to filter out error messages.
Not the best way to do it, but probably the easiest with the current setup we have for validation.